corehq/apps/hqwebapp/static/hqwebapp/js/key-value-mapping.js

```diff
@@ -138,3 +185,14 @@ uiElement.key_value_mapping = function (o) {
 };
 
 }());
+
+// To stack icon-upload modal over key-value modal
+// copied from http://stackoverflow.com/questions/19305821/multiple-modals-overlay
+$(document).on('show.bs.modal', '.modal', function () {
```

Comment:
This looks pretty fragile. Since you only have two modals and know how they're used, you don't need to support an infinite stack of modal like this code is doing. I think you could use event handlers and be at less risk for z-index problems:
- On or before the icon-upload modal is shown, close the key-value modal
- On or before the icon-upload modal is closed, re-open the key-value modal